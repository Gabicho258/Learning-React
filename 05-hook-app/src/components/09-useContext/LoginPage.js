import React, { useContext } from "react";
import { UserContext } from "./UserContext";

export const LoginPage = () => {
  const { setUser } = useContext(UserContext);
  const user = {
    id: 123,
    name: "Gabo",
  };
  return (
    <div>
      <h1>Login Page</h1>
      <hr />
      <button className="btn btn-primary" onClick={() => setUser(user)}>
        Login
      </button>
    </div>
  );
};
