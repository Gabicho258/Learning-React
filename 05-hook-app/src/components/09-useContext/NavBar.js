import React from "react";
import { Link } from "react-router-dom";
export const NavBar = () => {
  return (
    <nav className="navbar navbar-expand-sm navbar-dark bg-dark">
      <Link className="navbar-brand" to="/">
        useContext
      </Link>

      <div className="collapse navbar-collapse" id="navbarNavAltMarkup">
        <ul className="navbar-nav">
          <li className="nav-item nav-link">
            <Link to="/">Home</Link>
          </li>
          <li className="nav-item nav-link">
            <Link to="/about">About</Link>
          </li>
          <li className="nav-item nav-link">
            <Link to="/login">Login</Link>
          </li>
        </ul>
      </div>
    </nav>
  );
};
