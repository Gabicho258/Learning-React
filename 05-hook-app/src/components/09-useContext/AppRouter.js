import React from "react";
import { BrowserRouter as Router, Routes, Route } from "react-router-dom";
import { AboutPage } from "./AboutPage";
import { HomePage } from "./HomePage";
import { LoginPage } from "./LoginPage";
import { NavBar } from "./NavBar";

export const AppRouter = () => {
  return (
    <Router>
      <>
        <NavBar />
        <div className="container">
          <Routes>
            <Route path="/" element={<HomePage />} />
            <Route path="/about" element={<AboutPage />} />
            <Route path="/login" element={<LoginPage />} />
          </Routes>
        </div>
      </>
    </Router>
  );
};
