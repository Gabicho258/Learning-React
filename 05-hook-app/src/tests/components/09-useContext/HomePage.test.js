import React from "react";
import { HomePage } from "../../../components/09-useContext/HomePage";
import { mount } from "enzyme";
import { UserContext } from "../../../components/09-useContext/UserContext";

describe("Pruebas en <HomePage/>", () => {
  const user = {
    user: "Gabo",
    email: "test@test.com",
  };
  const wrapper = mount(
    <UserContext.Provider value={{ user }}>
      <HomePage />
    </UserContext.Provider>
  );
  test("debe de mostrarse el componente correctamente", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
