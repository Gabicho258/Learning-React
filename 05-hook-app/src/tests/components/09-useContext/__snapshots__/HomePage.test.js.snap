// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <HomePage/> debe de mostrarse el componente correctamente 1`] = `
<div>
  <h1>
    Home Page
  </h1>
  <hr />
  <pre>
    {
   "user": "Gabo",
   "email": "test@test.com"
}
  </pre>
</div>
`;
